### PR TITLE
Support more binary condition binding source

### PR DESCRIPTION
**Pull request by shrinktofit**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/animation/marionette/state-machine/condition/binding/auxiliary-curve-binding.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+/**
+ * @zh 一种过渡条件绑定，该绑定用于获取指定辅助曲线的当前值。该类绑定产生浮点值。
+ *
+ * @en A kind of transition condition binding,
+ * which is used to obtain the current value of specified auxiliary curve.
+ * This type of binding yields float value.
+ */
+@ccclass(`${CLASS_NAME_PREFIX_ANIM}TCAuxiliaryCurveBinding`)
+@menu('辅助曲线绑定')
```
Comment (SantyWang) [inline cocos/animation/marionette/state-machine/condition/binding/auxiliary-curve-binding.ts]: Needs i18n?

File: cocos/animation/marionette/state-machine/condition/binding/binding.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+/**
+ * @zh
+ * 描述过渡条件中，某个绑定提供的值的类型。
+ * @en
+ * Describes the type of value providing by a transition condition binding.
+ */
+export enum TCBindingValueType {
+    FLOAT = 0,
+
```
Comment (SantyWang) [inline cocos/animation/marionette/state-machine/condition/binding/binding.ts:11]: Is there a place holder?

File: cocos/animation/marionette/state-machine/condition/binding/auxiliary-curve-binding.ts (hunk 1 above)
Comment (shrinktofit) [reply cocos/animation/marionette/state-machine/condition/binding/auxiliary-curve-binding.ts]: Solved here: https://github.com/cocos/cocos-engine/pull/14857/commits/979f0ffae8f77f6847189ae6254f1b49f062b08f

File: cocos/animation/marionette/state-machine/condition/binding/binding.ts (hunk 2 above)
Comment (shrinktofit) [reply cocos/animation/marionette/state-machine/condition/binding/binding.ts:11]: In design, I wish the enumerators in `TCBindingValueType` to have number value equal to corresponding enumerator in `VariableType`. Due to typescript limit it's not currently possible to do:

```ts
enum TCBindingValueType { FLOAT = VariableType.FLOAT };

interface Map {
  [TCBindingValueType.FLOAT]: number; // Error here! `TCBindingValueType.FLOAT` can not be used as a property signature key!
}
```

So I have to duplicate its actual value.

However, I've document the intention on the constant value's side. See: https://github.com/cocos/cocos-engine/pull/14857/commits/979f0ffae8f77f6847189ae6254f1b49f062b08f#diff-e123ce10ac6a31e8c7cdb074c055046baac7942d98b98bb9e8eabd4b643bad41L9

**Review by SantyWang — APPROVED** [review]

**Pull request merged**